__tests__/10_deep_proxy_spec.ts

```diff
@@ -229,16 +229,52 @@ describe('special objects spec', () => {
     expect(() => { delete p1.c; }).toThrow();
   });
 
-  it('object with defineProperty (non-configurable and non-writable)', () => {
+  it('object with defineProperty (value, non-configurable & non-writeable)', () => {
     const proxyCache = new WeakMap();
-    const s1: any = {};
+    const s1: any = { c: 'c' };
     Object.defineProperty(s1, 'a', { value: { b: 'b' } });
     const a1 = new WeakMap();
     const p1 = createProxy(s1, a1, proxyCache);
     noop(p1.a.b);
     expect(isChanged(s1, s1, a1)).toBe(false);
     expect(isChanged(s1, { a: { b: 'b' } }, a1)).toBe(false);
     expect(isChanged(s1, { a: { b: 'b2' } }, a1)).toBe(true);
+    // Even though we've made a configurable copy, it is still non-writable, which is good b/c the
+    // new value would go to the internal proxyFriendlyCopy copy we'd made.
+    expect(() => {
+      p1.a = { b: 'b2' };
+    }).toThrowError("'set' on proxy: trap returned falsish for property 'a'");
+    // And because it is a copy, it is readonly for all properties
+    expect(() => {
+      p1.c = 'c2';
+    }).toThrowError("'set' on proxy: trap returned falsish for property 'c'");
+  });
+
+  it('object with defineProperty (non-configurable but writable)', () => {
+    const proxyCache = new WeakMap();
+    const s1: any = {};
+    Object.defineProperty(s1, 'a', { value: { b: 'b' }, configurable: false, writable: true });
+    const a1 = new WeakMap();
+    const p1 = createProxy(s1, a1, proxyCache);
+    noop(p1.a.b);
+  });
+
+  it('object with defineProperty (configurable but not-writable)', () => {
+    const proxyCache = new WeakMap();
+    const s1: any = {};
+    Object.defineProperty(s1, 'a', { value: { b: 'b' }, configurable: true, writable: false });
+    const a1 = new WeakMap();
+    const p1 = createProxy(s1, a1, proxyCache);
+    noop(p1.a.b);
+  });
+
+  it('object with defineProperty (non-configurable and not-writable)', () => {
+    const proxyCache = new WeakMap();
+    const s1: any = {};
+    Object.defineProperty(s1, 'a', { value: { b: 'b' }, configurable: false, writable: false });
```

Comment:
To make a difference from L232 test, should it be a getter?
```suggestion
    Object.defineProperty(s1, 'a', { get() { return { b: 'b' } }, configurable: false, writable: false });
```

Reply:
You're right that test L232 and test L274 both test `configurable: false` / `writable: false`; I wrote both just with the "black box" intent of "here's what happens if we just set value" (L232) and "here's what happens if we set both config: false, writable: false" (L274).

Granted, since we know (or more accurately I _now_ know that :-D) that "just setting value" effectively sets `config: false/writable: false`, they are in effect duplicate tests.

Personally I would probably keep both, just for a) keeping it more "black box" (I just like that style of testing), and b) as help for future newbie maintainers that don't immediately realize/remember they are actually the same behavior.

But also happy to have this 2nd L274 test just deleted. Or changed to the getter as you suggested.